runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeSceneOverlay.java

```diff
@@ -115,7 +115,11 @@ private void renderTiles(Graphics2D graphics, Player local)
 				LocalPoint location = rock.getLocalLocation();
 				if (localLocation.distanceTo(location) <= MAX_DISTANCE)
 				{
-					renderRock(graphics, rock);
+					// Only draw rockfalls on the same level
+					if (plugin.isUpstairs(localLocation) == plugin.isUpstairs(rock.getLocalLocation()))
```

Comment:
Combine the if conditions into one check:

```java
if (localLocation.distanceTo(location) <= MAX_DISTANCE
	&& plugin.isUpstairs(localLocation) == plugin.isUpstairs(rock.getLocalLocation()))
```

Reply:
@Nightfirecat Fixed, thanks.